components/brave_wallet_ui/common/async/lib.ts

```diff
@@ -34,7 +34,7 @@ import {
 import { getTokenParam, getFlattenedAccountBalances } from '../../utils/api-utils'
 import Amount from '../../utils/amount'
 import { sortTransactionByDate } from '../../utils/tx-utils'
-import { getBatTokensFromList, getNativeTokensFromList, getUniqueAssets } from '../../utils/asset-utils'
+import { getAssetIdKey, getBatTokensFromList, getNativeTokensFromList, getUniqueAssets } from '../../utils/asset-utils'
```

Comment:
I think the `getAssetIdKey` implementation needs updating (see my other comments as well):

**Before:**
```
export const getAssetIdKey = (asset: GetBlockchainTokenIdArg) => {
  return asset.isErc721
    ? `${asset.contractAddress}-${asset.tokenId}-${asset.chainId}`
    : `${asset.contractAddress}-${asset.chainId}`
}
```

**After:**
```
export const getAssetIdKey = (asset: GetBlockchainTokenIdArg) => {
  return asset.tokenId
    ? `${asset.contractAddress}-${asset.tokenId}-${asset.chainId}`
    : `${asset.contractAddress}-${asset.chainId}`
}
```

This looks better to me because it does not assume that tokens with `isErc721`, `isErc1155`, and `isNft` as `true` will have valid `tokenId` values.

Reply:
Also update `GetBlockchainTokenIdArg` accordingly.

Reply:
`SPL` NFT's do not have a `tokenId`.

Reply:
This has been updated!